queries.sql

```diff
@@ -87,13 +92,69 @@ COMMIT;
+
+/* List all animals owned by Dean Winchester that haven't tried to escape. */
+SELECT name, escape_attempts, full_name FROM animals JOIN owners ON animals.owner_id = owners.id WHERE escape_attempts = 0 AND full_name = 'Dean Winchester';
+
+/* Who owns the most animals? */
+SELECT full_name, COUNT(animals.owner_id) as total_animals FROM animals JOIN owners ON animals.owner_id = owners.id GROUP BY owners.id ORDER BY total_animals DESC LIMIT 1;
+SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31' GROUP BY species;
+
+SELECT animals.name, species.name FROM animals JOIN species ON animals.species_id = species.id WHERE species.name = 'Pokemon';
+/* Who owns the most animals? */
+SELECT full_name, COUNT(animals.owner_id) as total_animals FROM animals JOIN owners ON animals.owner_id = owners.id GROUP BY owners.id ORDER BY total_animals DESC LIMIT 1;
+
+SELECT animals.name, visits.date FROM animals JOIN visits ON animals.id=visits.animal_id JOIN vets ON visits.vet_id=vets.id 
+  WHERE vets.name='William Tatcher' GROUP BY animals.name, visits.date ORDER BY visits.date DESC LIMIT 1;
+SELECT COUNT(*) FROM animals JOIN visits ON animals.id=visits.animal_id JOIN vets ON visits.vet_id=vets.id 
+  WHERE vets.name='Stephanie Mendez';
+SELECT vets.name, species.name FROM vets 
+  LEFT JOIN specializations ON vets.id=specializations.vet_id 
+  LEFT JOIN species ON species.id=specializations.species_id
+  GROUP BY vets.name, species.name;
+SELECT animals.name, visits.date FROM animals JOIN visits ON animals.id=visits.animal_id JOIN vets ON visits.vet_id=vets.id 
+  WHERE vets.name='Stephanie Mendez' AND visits.date BETWEEN '2020-04-01' AND '2020-08-30' GROUP BY animals.name, visits.date;
+SELECT animals.name, COUNT(*) FROM animals JOIN visits ON animals.id=visits.animal_id GROUP BY animals.name ORDER BY COUNT(*) DESC LIMIT 1;
+SELECT animals.name, visits.date FROM animals JOIN visits ON animals.id=visits.animal_id JOIN vets ON visits.vet_id=vets.id 
+  WHERE vets.name='Maisy Smith'GROUP BY animals.name, visits.date ORDER BY visits.date ASC LIMIT 1;
+SELECT * FROM animals JOIN visits ON animals.id=visits.animal_id JOIN vets ON visits.vet_id=vets.id 
+  ORDER BY visits.date DESC LIMIT 1;
+select COUNT(vets.name)
+  from animals join visits on animals.id=visits.animal_id 
+                join vets on visits.vet_id=vets.id 
+                left join specializations on vets.id=specializations.vet_id
+  WHERE (SELECT specializations.species_id FROM specializations 
+          WHERE specializations.vet_id=visits.vet_id AND specializations.species_id=animals.species_id) IS NULL;
+SELECT animals.species_id, COUNT(*) FROM animals JOIN visits ON animals.id=visits.animal_id JOIN vets ON visits.vet_id=vets.id 
+  WHERE vets.name='Maisy Smith'GROUP BY animals.species_id; --Digimon
```

Comment:
## OPTIONAL 
* I strongly suggest that you comment on your queries so that it will be easy to know which one answers a particukar question.

Reply:
:green_circle: Great work on implementing these changes :+1: